### PR TITLE
Pass client timeout to server when accepting next session

### LINKED ISSUE #30962: [Service Bus] Respect TryTimeout when opening next available session link
The service hardcodes a max of 1 minute and 5 seconds when accepting a session. This means the TryTimeout that the user specifies in the client is ignored if they specify a value higher than this. We should specify the timeout property in the link settings properties when opening a link for next available session to support long polling. In the .NET library, we've found that using long polling results in better resource usage and avoids CPU spikes.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/30962

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs [hunk 1]
```diff
@@ -672,6 +679,29 @@ public override void Dispose()
                     OperationTimeout = _operationTimeout
                 };
 
+                if (isSessionReceiver && sessionId == null)
+                {
+                    // Subtract a random amount up to 100ms from the operation timeout as the jitter when attempting to open next available session link.
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs:684]: The commenting in this PR is exceptionally helpful!

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs [hunk 2]
```diff
@@ -672,6 +679,29 @@ public override void Dispose()
...
+                    var jitterBase = Math.Min(_operationTimeout.TotalMilliseconds / 100, OpenReceiveLinkBaseJitterMilliseconds);
+
+                    // We set the operation timeout on the properties not only to include the jitter, but also because the server will otherwise
+                    // restrict the maximum timeout to 1 minute and 5 seconds, regardless of the client timeout. We only do this for accepting next available
+                    // session as this is the only long-polling scenario.
+                    linkSettings.Properties = new Fields
+                    {
+                        {
+                            AmqpClientConstants.TimeoutName,
+                            (uint)_operationTimeout
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs:698]: So the service timeout is always shorter than the client?

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs (hunk 2 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs:698]: Yes, the service already subtracts 20ms off of their hardcoded 1 minute and 5 second timeout. This adds an additional buffer.

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs [hunk 3]
```diff
@@ -72,6 +72,13 @@ internal class AmqpConnectionScope : TransportConnectionScope
         /// </summary>
         private static int AuthorizationBaseJitterSeconds { get; } = 30;
 
+        ///// <summary>
```
Comment (conniey) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs:75]: I think there may have accidentally been more forward-slashes than you wanted

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs:75]: Ugh.. This is from when I was testing something and had commented out the new property.

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs:75]: https://github.com/Azure/azure-sdk-for-net/pull/30988

**Pull request merged**